codeclimate-validate-config/bin/run

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env ruby
+#
+# TODO: validate engine names as valid engines. this requires access to the
+# shared engine registry, so we should think carefully if that functionality is
+# worth the coupling.
+#
+###
+require "cc/yaml"
+require "forwardable"
+require "rainbow"
+
+module CC
+  class ValidateConfig
+    extend Forwardable
+
+    CODECLIMATE_YAML = ".codeclimate.yml".freeze
+
+    def_delegators :yaml,
+      :errors, :warnings, :nested_warnings
+
+    def initialize(strict)
+      @strict = strict
+
+      if File.exist?(CODECLIMATE_YAML)
+        @yaml = CC::Yaml.parse(File.read(CODECLIMATE_YAML))
+      else
+        @yaml = MissingConfig.new
+      end
+    end
+
+    def valid?
+      if strict?
+        [errors, warnings, nested_warnings].all?(&:empty?)
+      else
+        errors.empty?
+      end
+    end
+
+    private
+
+    attr_reader :yaml
+
+    def strict?
+      @strict
+    end
+
+    class MissingConfig
+      def errors
+        ["No '#{CODECLIMATE_YAML}' file found. Run 'codeclimate init' to generate a config file."]
+      end
+
+      def warnings; []; end
+      def nested_warnings; []; end
+    end
+  end
+end
+
+def colorize(text, *args)
+  @rainbow ||= Rainbow.new.tap do |rainbow|
+    rainbow.enabled = false unless $stdout.tty?
+  end
+
+  @rainbow.wrap(text).color(*args)
+end
+
+strict = %w[-s --strict].include?(ARGV.first)
```

Comment:
One of the biggest appeals of the isolated sub-commands is that they are free to make their own choices about dealing with whatever level of complexity they happen to be at. While `codeclimate` itself has long passed the bar for bringing in a real options parser (IMO), I don't think this command and its single flag meets it yet.

Also, this `--strict` idea might not even land (we might decide to preserve the "warnings are always fatal" behavior), so I didn't want to spend additional effort supporting it here.